### PR TITLE
Add check if EXERCISM_PRETTIER_VERSION matches prettier in package.json #1196

### PR DESCRIPTION
**Pull request by laulujan**
Issue #1196

Include check in the PR pre check to validate if ENV EXERCISM_PRETTIER_VERSION matches prettier in package.json, if not, set version in package.json to ENV.

### CODE REVIEW

**Comment by SleeplessByte** [comment]
@laulujan Great start. You could check out your change by changing `package.json` in the root to `2.3.0`. It should then log your message in the "pre-check" job.

However, this *updates* the environment to be "correct", but it won't use the correct version when running the format workflow. Instead of "fixing" it, it's probably best to blow up (error), and/or post a comment to the PR that the version mis-matches.

**Comment by laulujan** [comment]
Thanks, @SleeplessByte I committed a change that includes a message if the versions don't match, I also included a regex to compare only the version without the compatibility character, let me know if I understood correctly or if I still need to add something!

**Comment by SleeplessByte** [comment]
Great! 

Now all we need is _throwing an error_ instead of commenting, so that it _surfaces_ on PRs :). After you've made that change, we can try it out.

**Comment by laulujan** [comment]
Thanks @SleeplessByte I included the error in the new commit, let me know what you think!!

**Comment by SleeplessByte** [comment]
Alright. Onto the next thing. 

Currently it's checking the _current_ environment (so the environment of when pr.ci.yml runs). However, `EXERCISM_PRETTIER_VERSION` is only set on a step in `action-format.yml` and `verify-code-formatting.yml`. We therefore _can't_ rely on `env` to read this value.

What it needs to do is:

- Read out _both_ values (`action-format.yml` and `verify-code-formatting.yml`)
- Check each of those instances

The rest is looking good :)

**Comment by laulujan** [comment]
Thank you @SleeplessByte , I was wondering, is it ok if I  add https://github.com/nodeca/js-yaml as a dependency to read the yml files? Or do you know another approach I can take to read those files? Thanks in advance!!

**Comment by SleeplessByte** [comment]
Yeah, we can _guess_ here and read the source files _as text_ (using node's regular `fs`). Why? Well, yes, we _could_ parse the `.yml` file, but we are really looking for `EXERCISM_PRETTIER_VERSION: x.y.z`. We know that that doesn't appear anywhere in those files except for the "right place".

> "But @sleeplessbyte, what about the future?"

Good point. When someone else adds another instance of `EXERCISM_PRETTIER_VERSION`, for example as a comment, your check will throw an error, and it will be up to that person to fix it ;). 

So in short: read the file as text, search for the value by parsing it as text, for example using regular expressions.

**Comment by laulujan** [comment]
Great, thank you for your guidance, I included the changes in my latest commit, let me know what you think!

**Comment by SleeplessByte** [comment]
@laulujan I've converted your idea and combined it with changes we have in https://github.com/exercism/javascript-test-runner among other repos. Thank you so much for working on this, it was the basis of those changes.
